src/libappimage/desktop_integration/integrator/ResourcesExtractor.cpp

```diff
@@ -11,49 +11,114 @@ namespace appimage {
             ResourcesExtractor::ResourcesExtractor(const core::AppImage& appImage) : appImage(appImage) {}
 
             DesktopIntegrationResources ResourcesExtractor::extract() {
+                /* Build a cache of the entries contained in the AppImage payload. Include the entries path,
+                 * type and in case of a link link entry the link target. Solve links chains in order to ease
+                 * the lookup in the next step.
+                 */
+                PayloadEntriesCache entriesCache(appImage);
```

Comment:
Can't/Shouldn't this cache be, like, stored over the lifetime of this object? What if (in theory) I would call `extract()` more than once? Rebuilding a cache every time is totally unnecessary. Maybe it can be lazy-initialized as a class member (e.g., as shared pointer)?

Reply:
The resources extractor is meant to be used once per AppImage, as it will output all that's required for the desktop integration process. But thinking about it once more, it could be bounded to a single AppImage in order to store the cache internally and allow the extraction of random files in the future, also there will be no need for exposing that DesktopIntegrationResources struct that you hate so much :smile:

Reply:
Well, I'm always in for making managers manage a single object only.